### PR TITLE
Warning when pulling db changes - Issue #296

### PR DESCRIPTION
**Pull request by EduardoCopat**
Now warning when _pulling_ database changes into your local repository. I reused most of code from my last pull request.

### CODE REVIEW

File: src/zabapgit_objects.prog.abap [hunk 1]
```diff
@@ -763,6 +793,10 @@ CLASS lcl_objects_bridge IMPLEMENTATION.
 
   ENDMETHOD.                    "class_constructor
 
+  METHOD lif_object~compare_to_remote_version.
+
```
Comment (larshp) [inline src/zabapgit_objects.prog.abap]: create object missing here?

Comment (EduardoCopat) [inline src/zabapgit_objects.prog.abap]: Yes... I will commit an update. So places where the interfaces is used..

**Comment by larshp** [comment]
In this code:

```abap
      li_obj = create_object( is_item     = ls_item
                              iv_language = io_repo->get_master_language( )
                              is_metadata = lo_xml->get_metadata( ) ).

      READ TABLE lt_remote WITH KEY filename = <ls_result>-filename INTO ls_remote_file.
      IF ls_remote_file-filename NS '.abap'.

        "if file does not exist in remote, we don't need to validate
        IF sy-subrc = 0.
          CREATE OBJECT lo_current_version
            EXPORTING
              iv_xml = lcl_convert=>xstring_to_string_utf8( ls_remote_file-data ).
```

I think it hits too many files, the one we would like to check is the `|{ ls_item-obj_name }.{ ls_item-obj_type }.xml|`

perhaps rename lo_current_version to lo_remote version

**Comment by larshp** [comment]
or rather, perhaps just pass `lo_files` to the comparison, and let it find the correct files. That way we can extend the check to other files/objects if needed

**Comment by larshp** [comment]
if you copy+paste the code to https://larshp.github.io/abaplint/ and click issues, it should give the same result as from travis, that way it is possible to test vs abaplint before commiting

**Comment by EduardoCopat** [comment]
@larshp Thanks a lot for the inputs.

I refactored the code part you mentioned to be more efficient.

Also, I was running a travis CI on my fork, so abaplint was getting the issues. But when I commit to my branch in my fork. It automatically update here. (See: https://travis-ci.org/EduardoCopat/abapGit)

**Comment by larshp** [comment]
looks good to me, merging, thanks! :+1:

**Comment by EduardoCopat** [comment]
Great. Thanks for the reviews again. This commit was big because I had to add lots of code. I'll try to make the next ones smaller 👍

**Comment by larshp** [comment]
perhaps we need to look at 
https://github.com/larshp/abapGit/blob/master/src/zabapgit_objects_impl.prog.abap#L705
sometime in the future, would like to move the file decision down down to each object type.

But for now it will work :smile:

**Comment by EduardoCopat** [comment]
@larshp Yes. I did that because if we try to generate an XML out of an .abap, I received a short dump.

Sorry, my knowledge in the project is low yet.

**Comment by larshp** [comment]
no problem

I was thinking to have `lcl_object_tabl_validation` pick the correct file, in the future we might need to compare other files than the xml

**Pull request merged**